sources/rest_api/__init__.py

```diff
@@ -350,6 +358,60 @@ def paginate_dependent_resource(
     return resources
 
 
+def _validate_config(config: RESTAPIConfig) -> None:
+    c = deepcopy(config)
+    client_config = c.get("client")
+    if client_config:
+        auth = client_config.get("auth")
+        if auth:
+            auth = _mask_secrets(auth)
+
+    validate_dict(RESTAPIConfig, c, path=".")
+
+
+def _mask_secrets(auth_config: AuthConfig) -> AuthConfig:
+    if isinstance(auth_config, AuthBase) and not isinstance(
+        auth_config, AuthConfigBase
+    ):
+        return auth_config
+
+    has_sensitive_key = any(key in auth_config for key in SENSITIVE_KEYS)
+    if (
+        isinstance(auth_config, (APIKeyAuth, BearerTokenAuth, HttpBasicAuth))
+        or has_sensitive_key
+    ):
+        return _mask_secrets_dict(auth_config)
+    # Here, we assume that OAuth2 and other custom classes that don't implement __get__()
+    # also don't print secrets in __str__()
+    # TODO: call auth_config.mask_secrets() when that is implemented in dlt-core
+    return auth_config
+
+
+SENSITIVE_KEYS: List[str] = [
+    "token",
+    "api_key",
+    "username",
+    "password",
+]
+
+
+def _mask_secrets_dict(auth_config: AuthConfig) -> AuthConfig:
+    for sensitive_key in SENSITIVE_KEYS:
+        try:
+            auth_config[sensitive_key] = _mask_secret(auth_config[sensitive_key])  # type: ignore[literal-required, index]
+        except KeyError:
+            continue
+    return auth_config
+
+
+def _mask_secret(secret: Optional[str]) -> str:
+    if secret is None:
+        return "None"
+    if len(secret) < 3:
```

Comment:
Potential improvement is to make "3" a `MIN_SECRET_LENGTH = 3` module-level constant

Reply:
Oh, I wanted to have a <3 in the code, but it's better to refactor magic numbers to telling constant names